xbmc/dbwrappers/sqlitedataset.cpp

```diff
@@ -790,19 +793,19 @@ void SqliteDataset::make_deletion()
 void SqliteDataset::fill_fields()
 {
   //cout <<"rr "<<result.records.size()<<"|" << frecno <<"\n";
-  if ((db == NULL) || (result.record_header.empty()) ||
-      (result.records.size() < (unsigned int)frecno))
+  if (!db || (result.record_header.empty()) ||
+      (result.records.size() < static_cast<unsigned int>(frecno)))
     return;
 
   if (fields_object->empty()) // Filling columns name
   {
-    const unsigned int ncols = result.record_header.size();
+    const size_t ncols = result.record_header.size();
     fields_object->resize(ncols);
-    for (unsigned int i = 0; i < ncols; i++)
+    for (size_t i = 0; i < ncols; ++i)
     {
       (*fields_object)[i].props = result.record_header[i];
       std::string name = result.record_header[i].name;
-      name2indexMap.insert({str_toLower(name.data()), i});
+      name2indexMap.try_emplace(str_toLower(name.data()), i);
```

Comment:
raises build warnings of msvc because the map wants an int but i is size_t
```suggestion
      name2indexMap.try_emplace(str_toLower(name.data()), static_cast<int>(i));
```

Reply:
okay, will change that.